### PR TITLE
spectral-extraction: support for peak_method in auto/kosmos trace

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request extends on #1514 to add support for `peak_method` from https://github.com/astropy/specreduce/pull/115.  
Note that the auto/kosmos trace still often results in an error when actually extracting the spectrum (known bug in specreduce), so this review/PR should be isolated to seeing how the trace itself reacts to changes to the peak method when trace type is set to "Auto".

- [x] specreduce needs to be released and then that version pinned in jdaviz.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.vue [hunk 1]
```diff
@@ -85,6 +85,16 @@
           >
           </v-text-field>
         </v-row>
+
+        <v-row v-if="trace_type_selected==='Auto'">
+          <v-select
+            :items="trace_peak_method_items"
+            v-model="trace_peak_method_selected"
+            label="Peak Method"
+            hint="Method to use for identifying the peak in each bin"
```
Comment (ojustino) [inline jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.vue:94]: Can we be any more specific in the hint, or is "peak" enough on its own? I'm thinking about alternatives like "peak flux" (not always correct) or "peak signal."

**Review by ojustino — APPROVED** [review]
The code changes are sensible and I had no problems using the new dropdown. My comments are minor (and one isn't even about Jdaviz), so I'll give my approval.

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py [hunk 2]
```diff
@@ -397,8 +401,9 @@ def create_trace(self, add_data=False):
         elif self.trace_type_selected == 'Auto':
             trace = tracing.KosmosTrace(self.trace_dataset.selected_obj.data,
                                         guess=self.trace_pixel,
-                                        bins=self.trace_bins,
-                                        window=self.trace_window)
+                                        bins=int(self.trace_bins),
+                                        window=self.trace_window,
+                                        peak_method=self.trace_peak_method_selected.lower())
```
[the remark this answers is not in the document]
Comment (kecnry) [reply jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py:406]: I could also get around this by having the choices in jdaviz be a dictionary (labels for the dropdowns vs values actually passed to specreduce), but that felt like it's just unnecessarily unwieldy.  Specreduce should raise a helpful error message if you pass an invalid string, so for users calling the API manually, I'm not sure the casting in specreduce is necessary... but definitely something we can consider!

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.vue (hunk 1 above)
Comment (kecnry) [reply jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.vue:94]: "peak pixel"?

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ojustino) [reply jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py:406]: Calling `self.peak_method = self.peak_method.lower()` in `KosmosTrace` doesn't have to preclude a check that the string is valid and wouldn't cost much. I agree that making the dictionary on the Jdaviz side is too much.

**Comment by ojustino** [comment]
The merge conflict happened after my approval, so I'll check again once its been resolved.

**Review by rosteen — APPROVED** [review]
The new selection seems to work for me, I can see the trace change depending on the choice. I can only assume that the underlying math in specreduce is right, out of scope for this review.

**Pull request merged**